internal/services/appservice/source_control_slot_resource.go

```diff
@@ -0,0 +1,307 @@
+				}
+
+				usesLinux := false
+				if strings.Contains(strings.ToLower(*app.Kind), "linux") {
+					usesLinux = true
+				}
+
+				sourceControl := web.SiteSourceControl{
+					SiteSourceControlProperties: &web.SiteSourceControlProperties{
+						IsManualIntegration:       utils.Bool(appSourceControlSlot.ManualIntegration),
+						DeploymentRollbackEnabled: utils.Bool(appSourceControlSlot.RollbackEnabled),
+						IsMercurial:               utils.Bool(appSourceControlSlot.UseMercurial),
+					},
+				}
+
+				if appSourceControlSlot.RepoURL != "" {
+					sourceControl.SiteSourceControlProperties.RepoURL = utils.String(appSourceControlSlot.RepoURL)
+				}
+
+				if appSourceControlSlot.Branch != "" {
+					sourceControl.SiteSourceControlProperties.Branch = utils.String(appSourceControlSlot.Branch)
+				}
+
+				if ghaConfig := expandGithubActionConfig(appSourceControlSlot.GithubActionConfiguration, usesLinux); ghaConfig != nil {
+					sourceControl.SiteSourceControlProperties.GitHubActionConfiguration = ghaConfig
+				}
+
+				_, err = client.UpdateSourceControlSlot(ctx, id.ResourceGroup, id.SiteName, sourceControl, id.SlotName)
+				if err != nil {
+					return fmt.Errorf("creating Source Control configuration for %s: %v", id, err)
+				}
+			}
+
+			// TODO - Need to introduce polling for deployment statuses to avoid 409's elsewhere
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r SourceControlSlotResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			id, err := parse.WebAppSlotID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := metadata.Client.AppService.WebAppsClient
+
+			appSourceControl, err := client.GetSourceControlSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName)
+			if err != nil || appSourceControl.SiteSourceControlProperties == nil {
+				if utils.ResponseWasNotFound(appSourceControl.Response) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading Source Control for %s: %v", id, err)
+			}
+
+			siteConfig, err := client.GetConfigurationSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName)
+			if err != nil {
+				return fmt.Errorf("reading App for Source Control %s: %v", id, err)
+			}
+
+			if siteConfig.ScmType == web.ScmTypeNone {
+				metadata.Logger.Infof("App %s SCMType is `None` removing Source Control resource from state", id.SiteName)
+				metadata.ResourceData.SetId("")
+			}
+
+			props := *appSourceControl.SiteSourceControlProperties
+
+			state := SourceControlSlotModel{
+				SlotID:                    id.ID(),
+				SCMType:                   string(siteConfig.ScmType),
+				RepoURL:                   utils.NormalizeNilableString(props.RepoURL),
+				Branch:                    utils.NormalizeNilableString(props.Branch),
+				ManualIntegration:         *props.IsManualIntegration,
```

Comment:
Any chance of these and the next three lines being nil?

Reply:
oh, good spot, thought I'd caught all those!